cmd/airflow.go

```diff
@@ -122,6 +124,11 @@ func newDevRootCmd(platformCoreClient astroplatformcore.CoreClient, astroCoreCli
 		Aliases: []string{"d"},
 		Short:   "Run your Astro project in a local Airflow environment",
 		Long:    "Run an Apache Airflow environment on your local machine to test your project, including DAGs, Python Packages, and plugins.",
+		// Most astro dev sub-commands require the container runtime,
+		// so we set that configuration in this persistent pre-run hook.
+		// A few sub-commands don't require this, so they explicitly
+		// clobber it with a no-op function.
+		PersistentPreRunE: ConfigureContainerRuntime,
```

Comment:
Here we insert a new `PersistentPreRunE` function to setup the runtime for all the sub commands. This is a quick function and doesn't actually run any initialization steps that take time. When needed, the value set from this function is referenced and we run any lifecycle hooks that interact with the runtime.

With this PR, we're only running this on the `astro dev start` command, as the other commands explicitly clobber this hook. In the following PR for podman support, we'll utilize this much more. 

The methods that are not yet using this functionality explicitly define their own `PersistentPreRunE` function of `DoNothing`. This was already happening previously with an inline function definiton. This is really just cleaning that up and defining all our hooks in a single place.